bluesky/run_engine.py

```diff
@@ -464,9 +464,9 @@ def __call__(self, plan, subs=None, **metadata):
 
         # Register temporary subscriptions. Save tokens to unsubscribe later.
         subs = normalize_subs_input(subs)
-        try:
+        if hasattr(plan, 'subs'):
```

Comment:
This part of the code does not have to be high-perf, and this pattern avoids accidentally snarfing AttributeErrors that might bubble up from `normalize_subs_input` or something. The right way to do this is to put that in else, but at that point the code just seems obfuscated. I like this better.